huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go

```diff
@@ -87,53 +91,53 @@ func resourceComputeVolumeAttachCreate(d *schema.ResourceData, meta interface{})
 		ServerId: instanceId,
 	}
 
-	logp.Printf("[DEBUG] Creating volume attachment: %#v", attachOpts)
+	log.Printf("[DEBUG] Creating volume attachment: %#v", attachOpts)
 	job, err := block_devices.Attach(computeClient, attachOpts)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	logp.Printf("[DEBUG] The response of volume attachment request is: %#v", job)
+	log.Printf("[DEBUG] The response of volume attachment request is: %#v", job)
 	stateConf := &resource.StateChangeConf{
-		Pending:    []string{"INIT", "RUNNING"},
-		Target:     []string{"SUCCESS"},
-		Refresh:    AttachmentJobRefreshFunc(computeClient, job.ID),
-		Timeout:    d.Timeout(schema.TimeoutCreate),
-		Delay:      30 * time.Second,
-		MinTimeout: 15 * time.Second,
+		Pending:      []string{"INIT", "RUNNING"},
+		Target:       []string{"SUCCESS"},
+		Refresh:      AttachmentJobRefreshFunc(computeClient, job.ID),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        30 * time.Second,
+		PollInterval: 15 * time.Second,
 	}
 
-	if _, err = stateConf.WaitForState(); err != nil {
-		return fmtp.Errorf("Error attaching HuaweiCloud volume: %s", err)
+	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+		return diag.Errorf("Error attaching volume: %s", err)
 	}
 
 	// The old logic use the instance ID and attachment ID as the resource ID, and the attachment ID equals
 	// the volume ID.
 	id := fmt.Sprintf("%s/%s", instanceId, volumeId)
 	d.SetId(id)
 
-	return resourceComputeVolumeAttachRead(d, meta)
+	return resourceComputeVolumeAttachRead(ctx, d, meta)
 }
 
-func resourceComputeVolumeAttachRead(d *schema.ResourceData, meta interface{}) error {
+func resourceComputeVolumeAttachRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
suggest to use "_" if the input parameter was not used

Reply:
ok